firehose/client.go

```diff
@@ -62,13 +63,20 @@ func (c *Client) Start() {
 		}
 	}
 
-	subscriptionID := c.options.SubscriptionID
-	if len(subscriptionID) == 0 {
-		subscriptionID = "FirehosePlugin"
+	var errors <-chan error
+	var output <-chan *events.Envelope
+	if len(c.options.AppGUID) != 0 {
```

Comment:
@jtuchscherer Interesting that you used `len` here vs `c.options.AppGUID != ""`. Is there a reason for this? Reads a bit weird.